monitoring.go

```diff
@@ -297,14 +297,31 @@ func monitoring(
 ) {
 	defer close(resp)
 
-	ctx, cancel := context.WithTimeout(context.Background(), sc.rest.RequestTimeout)
+	ctx, cancel := context.WithTimeout(context.Background(), sc.cfg.MonitoringFetcher.MaxDuration)
 	defer cancel()
 
-	var m monitoringResponse
-	err := sc.getMonitoringResult(ctx, "queries", qid, &m)
-	if err == nil && len(m.Data.Queries) == 1 {
-		resp <- &m.Data.Queries[0]
+	var queryMonitoringData *QueryMonitoringData
+	for {
```

Comment:
I don't think we want this to run forever... we should have some sort of hard time limit.

Reply:
It's not forever. the actual call gets a context with a deadline. It will fail with that deadline and exit

Reply:
Doh, good point. I haven't written golang in a while 🤦‍♂️